### PR TITLE
Add shipping rates import through YAML and integration test

### PR DESCRIPTION
**Pull request by ghost**
I have added the sample file to the main master.yml and tested it on a clean magento install.

To run the integration test, follow instructions in the README.

### CODE REVIEW

File: Model/Component/ShippingTableRates.php [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+        /** @var Region */
+        $regionModel = $this->regionFactory->create();
+        $regionModel = $regionModel->loadByCode($shippingRate['dest_region_code'], $shippingRate['dest_country_id']);
+        $regionId = $regionModel->getId();
+
+        $this->removeYamlKeysFromDatabaseInsert($shippingRate);
+
+        $shippingRate = array_merge(array('dest_region_id' => $regionId), $shippingRate);
+        $shippingRate = array_merge(array('website_id' => $websiteId), $shippingRate);
```
Comment (rgpjones) [inline Model/Component/ShippingTableRates.php]: You could use short array notation here (i.e.. `[ ]`)

File: Model/Component/ShippingTableRates.php [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+        $this->log->logInfo(
+            sprintf(
+                "Shipping rate #%s for website %s being created",
+                ++$shippingRateCount,
+                $website
+            )
+        );
+        $tablerateModel->getConnection()
+            ->insertArray($tablerateModel->getMainTable(), $columns, array($shippingRate));
```
Comment (rgpjones) [inline Model/Component/ShippingTableRates.php]: and also here

File: Model/Component/ShippingTableRates.php (hunk 1 above)
Comment (rgpjones) [reply Model/Component/ShippingTableRates.php]: You could rewrite this a bit, as
$shippingRate = array_merge(
    [
        'dest_region_id' => $regionId,
        'website_id' => $websiteId
    ],
    $shippingRate
);

**Review by rgpjones — APPROVED** [review]
A couple of style changes you might want to make, but this largely looks okay to me.

File: Model/Component/ShippingTableRates.php (hunk 1 above)
Comment (ghost) [reply Model/Component/ShippingTableRates.php]: fixed

File: Model/Component/ShippingTableRates.php (hunk 2 above)
Comment (ghost) [reply Model/Component/ShippingTableRates.php]: done

**Pull request merged**